build.sbt

```diff
@@ -42,3 +46,30 @@ releaseProcess := Seq[ReleaseStep](
   commitNextVersion,
   pushChanges
 )
+
+ThisBuild / githubWorkflowBuild := Seq(WorkflowStep.Sbt(List("test", "scripted")))
+
+ThisBuild / githubWorkflowTargetTags ++= Seq("v*")
+// Remove this when we set up publishing via sbt-ci-release
+ThisBuild / githubWorkflowPublishTargetBranches := Seq()
+
+ThisBuild / githubWorkflowOSes := Seq("ubuntu-latest", "macos-latest", "windows-latest")
+
+ThisBuild / githubWorkflowJavaVersions := Seq(
+  JavaSpec.temurin("8"),
+  JavaSpec.temurin("11"),
+  JavaSpec.temurin("17"),
+  JavaSpec.temurin("21")
+)
+
+// Necessary to setup git so that sbt-scripted tests can run on github actions
+ThisBuild / githubWorkflowBuildPreamble := Seq(
+  WorkflowStep.Run(
+    commands = List(
+      "git config --global init.defaultBranch main",
+      """git config --global user.email "sbt-pull-request-validator@github.com"""",
```

Comment:
So I honestly don't have an idea of what is appropriate to put here, I just need to put something otherwise the sbt-scripted tests fail since git isn't initialized in the correct state (the sbt-scripted tests manually use git to create commits in order to test the project properly).